pkg/ccl/streamingccl/streamingest/stream_ingestion_polling_processor.go

```diff
@@ -0,0 +1,165 @@
+
+type streamIngestionPollingProcessor struct {
+	execinfra.ProcessorBase
+	execinfra.StreamingProcessor
+
+	flowCtx *execinfra.FlowCtx
+	spec    execinfrapb.StreamIngestionPollingSpec
+	output  execinfra.RowReceiver
+
+	jobID      int64
+	registry   *jobs.Registry
+	stopPoller chan struct{}
+}
+
+var _ execinfra.Processor = &streamIngestionPollingProcessor{}
+var _ execinfra.RowSource = &streamIngestionPollingProcessor{}
+
+func init() {
+	rowexec.NewStreamIngestionPollingProcessor = newStreamIngestionPollingProcessor
+}
+
+func newStreamIngestionPollingProcessor(
+	flowCtx *execinfra.FlowCtx,
+	processorID int32,
+	spec execinfrapb.StreamIngestionPollingSpec,
+	post *execinfrapb.PostProcessSpec,
+	output execinfra.RowReceiver,
+) (execinfra.Processor, error) {
+	sp := &streamIngestionPollingProcessor{
+		flowCtx:    flowCtx,
+		spec:       spec,
+		output:     output,
+		jobID:      spec.JobID,
+		registry:   flowCtx.Cfg.JobRegistry,
+		stopPoller: make(chan struct{}),
+	}
+	if err := sp.Init(
+		sp,
+		post,
+		streamIngestionResultTypes,
+		flowCtx,
+		processorID,
+		output,
+		nil, /* memMonitor */
+		execinfra.ProcStateOpts{
+			InputsToDrain: []execinfra.RowSource{},
+			TrailingMetaCallback: func(ctx context.Context) []execinfrapb.ProducerMetadata {
+				sp.close()
+				return nil
+			},
+		},
+	); err != nil {
+		return nil, err
+	}
+	return sp, nil
+}
+
+func (sp *streamIngestionPollingProcessor) close() {
+	if sp.InternalClose() {
+		close(sp.stopPoller)
+	}
+}
+
+// Start is part of the RowSource interface.
+func (sp *streamIngestionPollingProcessor) Start(ctx context.Context) context.Context {
+	return sp.StartInternal(ctx, streamIngestionPollingProcName)
+}
+
+// checkForCutoverSignal periodically loads the job progress to check for the
+// sentinel value that signals the ingestion job to complete.
+func (sp *streamIngestionPollingProcessor) checkForCutoverSignal(
+	ctx context.Context,
+) (rowenc.EncDatumRow, error) {
+	tick := time.NewTicker(time.Second * 10)
+	defer tick.Stop()
+	for {
+		select {
+		case _, ok := <-sp.stopPoller:
+			if !ok {
+				return nil, nil
+			}
+			// Shouldn't come here.
+			return nil, nil
+		case <-tick.C:
+			j, err := sp.registry.LoadJob(ctx, sp.jobID)
+			if err != nil {
+				return nil, err
+			}
+			progress := j.Progress()
+			var prog *jobspb.Progress_StreamIngest
+			var ok bool
+			if prog, ok = progress.GetDetails().(*jobspb.Progress_StreamIngest); !ok {
+				return nil, errors.Newf("unknown progress type %T in stream ingestion job %d",
+					j.Progress().Progress, sp.jobID)
+			}
+			// Job has been signaled to complete.
+			if cutoverTime := prog.StreamIngest.CutoverTime; !cutoverTime.IsEmpty() {
+				streamIngestBytes, err := protoutil.Marshal(prog.StreamIngest)
+				if err != nil {
+					return nil, err
+				}
+				row := rowenc.EncDatumRow{
+					rowenc.DatumToEncDatum(types.Bytes, tree.NewDBytes(tree.DBytes(streamIngestBytes))),
+				}
+				return row, nil
```

Comment:
After we signal, what is the correct behavior? Is it okay that the processor might push this same row tick seconds later? I also played around with the idea of immediately tearing down the processor once we push a row but a downstream stream ingestion processor might handle the meta before the signalled row and that doesn't seem like what we want.

Currently, the stream ingestion processor will call `ConsumerClosed()` and then teardown this processor.